pkgs/jni/test/jarray_test.dart

```diff
@@ -48,24 +60,44 @@ void run({required TestRunnerCallback testRunner}) {
     using((arena) {
       final array = JArray(jchar.type, 3)..releasedBy(arena);
       expect(array.length, 3);
-      array[0] = 'ح';
-      array[1] = '2';
-      array[2] = '3';
-      expect(array[0], 'ح');
-      expect(array[1], '2');
-      expect(array[2], '3');
-      array.setRange(0, 3, ['4', '5', '6', '7'], 1);
-      expect(array[0], '5');
-      expect(array[1], '6');
-      expect(array[2], '7');
+      array[0] = 'ح'.codeUnitAt(0);
+      array[1] = '2'.codeUnitAt(0);
+      array[2] = '3'.codeUnitAt(0);
```

Comment:
Please add a test writes the UTF16 code units from Dart, then converts the char array to a string in Java and then read the String again in Dart as a JString to see that the conversions work as expected. And ensure there are some characters in there that are more than one code unit. 😄